openlibrary/core/bookshelves.py

```diff
@@ -346,13 +432,32 @@ def get_filtered_reading_log_books(
                 facet=False,
                 # Putting these in fq allows them to avoid user-query processing, which
                 # can be (surprisingly) slow if we have ~20k OR clauses.
-                extra_params=[('fq', f'key:({" OR ".join(reading_log_work_keys)})')],
+                extra_params=[('fq', filter_query)],
             )
+
             total_results = solr_resp.num_found
+            solr_docs = solr_resp.docs
+
+            if editions:
+                matching_editions = [
+                    logged_work_edition_dict.get(work["key"])
+                    for work in solr_docs
+                    if logged_work_edition_dict.get(work["key"])
+                ]
+
+                edition_data = get_solr().get_many(
+                    [f'{key}' for key in matching_editions],
```

Comment:
```suggestion
                    matching_editions,
```

Reply:
This was extremely helpful, and made it into the final version.